#! /bin/sh /usr/share/dpatch/dpatch-run
## 03_fix_class_mapping.dpatch by  <cajus@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix location of auto-generated class mapping file

@DPATCH@
diff -urNad gosa-2.6.3~/include/class_config.inc gosa-2.6.3/include/class_config.inc
--- gosa-2.6.3~/include/class_config.inc	2008-12-18 15:13:44.000000000 +0100
+++ gosa-2.6.3/include/class_config.inc	2009-01-15 14:08:54.000000000 +0100
@@ -76,12 +76,12 @@
         if we have installed or removed plugins. 
      */
     if(session::global_is_set("class_location.inc:timestamp")){
-      $tmp = stat("../include/class_location.inc");
+      $tmp = stat("/var/cache/gosa/class.cache");
       if($tmp['mtime'] != session::global_get("class_location.inc:timestamp")){
         session::global_un_set("plist");
       }
     }
-    $tmp = stat("../include/class_location.inc");
+    $tmp = stat("/var/cache/gosa/class.cache");
     session::global_set("class_location.inc:timestamp",$tmp['mtime']);
 
 
diff -urNad gosa-2.6.3~/include/functions.inc gosa-2.6.3/include/functions.inc
--- gosa-2.6.3~/include/functions.inc	2009-01-15 14:07:35.000000000 +0100
+++ gosa-2.6.3/include/functions.inc	2009-01-15 14:07:36.000000000 +0100
@@ -73,7 +73,7 @@
 $svn_revision = '$Revision: 13322 $';
 
 /* Include required files */
-require_once("class_location.inc");
+require_once("/var/cache/gosa/class.cache");
 require_once ("functions_debug.inc");
 require_once ("accept-to-gettext.inc");
 
diff -urNad gosa-2.6.3~/update-gosa gosa-2.6.3/update-gosa
--- gosa-2.6.3~/update-gosa	2008-06-10 09:56:22.000000000 +0200
+++ gosa-2.6.3/update-gosa	2009-01-15 14:07:36.000000000 +0100
@@ -21,7 +21,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-define ("GOSA_HOME", dirname(__FILE__));
+define ("GOSA_HOME", "/usr/share/gosa");
 define ("LOCALE_DIR", GOSA_HOME."/locale");
 define ("PLUGSTATE_DIR", GOSA_HOME."/state");
 
@@ -122,8 +122,8 @@
 function rescan_classes()
 {
 	echo "Updating class cache...\n";
-	$class_mapping= get_classes();
-	$filename= GOSA_HOME."/include/class_location.inc";
+	$class_mapping= get_classes("/usr/share/gosa");
+	$filename= "/var/cache/gosa/class.cache";
 
 	/* Sanity checks */
 	if (!file_exists($filename) || is_writable($filename)) {
